src/check-update.ts

```diff
@@ -0,0 +1,250 @@
+        update,
+        breaking,
+        error,
+      };
+    }
+  }
+
+  return versionMap;
+}
+
+/** readProjectDependencies reads from possible dependency files
+ * (import_map.json, slack.json) and maps them to the versionMap
+ * containing each dependency's update information
+ */
+async function readProjectDependencies(): Promise<VersionMap> {
+  const cwd = Deno.cwd();
+  const versionMap: VersionMap = {};
+
+  // Find SDK component versions in import map, if available
+  const map = await getJson(`${cwd}/import_map.json`);
+  for (const sdkUrl of Object.values(map.imports) as string[]) {
+    for (const sdk of IMPORT_MAP_SDKS) {
+      if (sdkUrl.includes(sdk)) {
+        versionMap[sdk] = {
+          latest: "",
+          update: false,
+          breaking: false,
+          error: "",
+          current: extractVersion(sdkUrl),
+        };
+      }
+    }
+  }
+
+  // Find SDK component versions in slack.json, if available
+  const { hooks }: { [key: string]: string } = await getJson(
+    `${cwd}/slack.json`,
+  );
+  for (const command of Object.values(hooks)) {
+    // TODO :: Does not cover the case where multiple commands use the same SDK.
+    // Only the "last" key containing the relevant SDK will be checked as-is
+    for (const sdk of SLACK_JSON_SDKS) {
+      if (command.includes(sdk)) {
+        versionMap[sdk] = {
+          latest: "",
+          update: false,
+          breaking: false,
+          error: "",
+          current: extractVersion(command),
+        };
+      }
+    }
+  }
+
+  return versionMap;
+}
+
+async function getJson(file: string) {
+  return JSON.parse(await Deno.readTextFile(file));
+}
+
+/** fetchLatestModuleVersion makes a call to deno.land with the
+ * module name and returns the extracted version number, if found
+ */
+async function fetchLatestModuleVersion(moduleName: string): Promise<string> {
+  const res = await fetch(`https://deno.land/x/${moduleName}`, {
+    redirect: "manual",
+  });
+
+  const redirect = res.headers.get("location");
+  if (redirect === null) {
+    throw new Error(`${moduleName} not found on deno.land!`);
+  }
+
+  return extractVersion(redirect);
+}
+
+function extractVersion(str: string) {
+  const at = str.indexOf("@");
+  const slash = str.indexOf("/", at);
+  const version = slash < at
+    ? str.substring(at + 1)
+    : str.substring(at + 1, slash);
+  return version;
+}
+
+/** createDependencyMsg  creates a terminal-friendly display message
+ * that features the name of the dependency, current and latest versions,
+ * and visual indication if the change is breaking or an error occurred:
+ *
+ * › deno_slack_sdk
+ *   0.0.3 → 0.0.4
+ */
+function createDependencyMsg(
+  sdk: string,
+  breaking: boolean,
+  current: string,
+  latest: string,
+  error: string,
+): string {
+  const red = "\x1b[31m";
+  const yellow = "\x1b[38;5;214m";
+  const blue = "\x1b[38;5;39m";
+  const grey = "\x1b[38;5;243m";
+  const reset = "\x1b[0m";
+  const bold = "\x1b[1m";
```

Comment:
My first foray into ANSI escape codes, so if I'm doing it wrong, please do let me know!